crates/ruff_python_formatter/src/expression/binary_like.rs

```diff
@@ -317,46 +317,58 @@ impl Format<PyFormatContext<'_>> for BinaryLike<'_> {
                     //     ^^^^^^ this part or ^^^^^^^ this part
                     // ```
                     if let Some(left_operator_index) = index.left_operator() {
-                        // Everything between the last implicit concatenated string and the left operator
-                        // right before the implicit concatenated string:
+                        // Handles the case where the left and right side of a binary expression are both
+                        // implicit concatenated strings. In this case, the left operator has already been written
+                        // by the preceding implicit concatenated string. It is only necessary to finish the group,
+                        // wrapping the soft line break and operator.
```

Comment:
I don't know if it matters, but doesn't this mean that the right side of `"a" "b" + "c" "d"` will be formatted using the right-side formatting logic, which doesn't include the `with_layout(StringLayout::ImplicitConcatenatedStringInBinaryLike)`?

Reply:
The implicit string formatting formats `left_operand` (optional) `left_operator` `implicit_concat` `right_operator`. 

So for, `a + "b" "c" * "d" "e"` it formats `a`, `+`, `"b" "c"` and `*`. The right side of `*` gets formatted as part of the next implicit concatenated string.

https://github.com/astral-sh/ruff/blob/1e6df19a351170ded4b1e1a4e068993107bd1047/crates/ruff_python_formatter/src/expression/binary_like.rs#L393-L431

Reply:
Ohh it only formats right _operator_, okay, got it, thank you. This is what was blocking me last night :D

Reply:
Yes, it only formats the right operator, but starts a group that includes the right operand (without formatting it)